h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java

```diff
@@ -152,6 +153,41 @@ public static void maybePrintYarnLogsMessage() {
     maybePrintYarnLogsMessage(true);
   }
 
+  private static class PortRange {
+    int from;
+    int to;
+
+    public PortRange(int from, int to) {
+      this.from = from;
+      this.to = to;
+    }
+
+    void validate() {
+      if (from > to)
+        error("Invalid port range (lower bound larger than upper bound: " + this + ").");
+      if ((from == 0) && (to != 0))
+        error("Invalid port range (lower bound cannot be 0).");
+      if (to > 65535)
+        error("Invalid port range (upper bound > 65535).");
+    }
+
+    boolean isSinglePort() {
+      return from == to;
+    }
+
+    static PortRange parse(String rangeSpec) {
+      if (rangeSpec == null)
+        throw new NullPointerException("Port range is not specified (null).");
+      String[] ports = rangeSpec.split("-");
+      if (ports.length != 2)
+        throw new IllegalArgumentException("Invalid port range specification (" + rangeSpec + ")");
+      return new PortRange(Integer.parseInt(ports[0]), Integer.parseInt(ports[1]));
```

Comment:
Minor: I would do "ports[0].trim" to simplify user experience (e.g., to handle `"12345 - 23456"`)